### DISCUSSION TITLE
How to customize a shortcut key for a third-party reader？

### DISCUSSION THREAD

**Discussion by tyf2018** [sumatrapdfreader/sumatrapdf#2991] [Q&A]
I used ExternalViewers to specify a third-party reader, which works well. Now I want to customize a shortcut key for this purpose. What should I do?

I checked the instructions. It seems that custom shortcut keys only support internal commands?

Best regards.

**Comment by GitHubRulesOK** [comment]
the ampersand in the name allows for 3 finger salute e.g. in english it will be

For [Open in] 
name=My Extra &View
<kbd>Alt</kbd><kbd>F</kbd><kbd>V</kbd>
the F is for File menu and the &V for  the desired choice

options are limited to ones not already in use, but if duplicated then needs more choice up down or enter

I think this desire was mentioned already so if found this one might be tagged duplicate.

**Reply by tyf2018** [reply]
@GitHubRulesOK 

Thanks for your answer, but I don't quite understand your setting method.
I want to bind a shortcut key with a rule similar to the following，but I don't know what content should be filled in the cmd.

```
Shortcuts [
	[
		Cmd = CmdOpenWithAcrobat
		Key = Ctrl + T
	]
]
```

**Reply by GitHubRulesOK** [reply]
if its Acrobat is installed the menu should show "Open in &Adobe Reader" (<kbd>Alt</kbd><kbd>F</kbd><kbd>A</kbd>)
and to use a non installed version you would historically replace/emulate with
ExternalViewers [
blah blah
]

Without a current Acrobat installed I  do not know if "CmdOpenWithAcrobat" should work so have to ask
 is Acrobat showing in the File Menu by default as detected ? 
If not then that combination may not be assigned.

**Comment by tyf2018** [comment]
I think I should not have made it clear, and for the avoidance of ambiguity, I will now give a complete picture of the actual situation.

I use the portable version of PDFXEdit .exe and work well using the rules described below.

```
ExternalViewers [
	[
		CommandLine = "F:\Portable_Update\PDF-XChange\PDFXEdit_Main.exe" /A page=%p "%1"
		Name = PDFXEdit_Main【便携版】
		Filter = *.pdf
		Key = Ctrl + E
	]
]
```
![CopyQ QRGhVC](https://user-images.githubusercontent.com/56342788/193378564-b3462ac5-fbac-409e-aa8f-6093cc804465.png)

The only small problem now is that I want to add a shortcut key to the portable version of PDFXEdit .exe, how should I add it? I consulted the user manual, and it seems that custom shortcuts only support internal commands like "CmdOpenWithPdfXchange", but not for the portable version.

Best regards.

**Reply by GitHubRulesOK** [reply]
I would agree that's a good analysis the test is simply
 if exist C:\Program Files\Tracker Software\PDF Editor\PDFXEdit.exe then CmdOpenWithPdfXchange so any exe at that location seems to do the trick, if I put a mgsbox.exe as that pdfedit.exe and press t then I get good feedback.

![image](https://user-images.githubusercontent.com/12754913/193385833-2ebace5e-d122-4667-9b3d-7ba8fac27493.png)
